__test__/rest/user.test.js

```diff
@@ -0,0 +1,81 @@
+const request = require('supertest');
+const app = require('../../app');
+const mongoose = require('mongoose');
+const {settings, dbURI} = require('../../config/db');
+const User = require('../../models/User');
+const faker = require('faker');
+
+
+describe('some initial tests for user', () => {
+    //global variables that we can use in all tests
+    let randomUser, registerResponse, connection;
+
+    beforeAll(async done =>{
+        //waiting for connection with mongoDB database
+        connection = await mongoose.connect(dbURI, settings);
+        done()
+    });
+    test('should create new user', async () => {
+        //create user with random properties.
+        const userToRegister = {
+            firstName: faker.name.firstName(),
+            lastName: faker.name.lastName(),
+            email: faker.internet.email(),
+            password: faker.internet.password()
+        };
+
+        const response = await request(app).post('/api/users/register')
+            .type('form')
+            .send({...userToRegister})
+            .set('Accept', 'application/json');
+        expect(response.body.msg).toContain('registered');
+        expect(response.body.token).toContain('Bearer');
+
+        const user = await User.findOne({email: userToRegister.email});
+        expect(user).toHaveProperty('firstName',userToRegister.firstName);
+        expect(user).toHaveProperty('lastName',userToRegister.lastName);
+        expect(user).toHaveProperty('email',userToRegister.email);
+        expect(user).toHaveProperty('emailVerified',false)
+
+    });
+    describe('create one random user for all future tests', () => {
+        // EVERY TEST HERE WILL OPERATE ON SAME USER!
+        beforeAll(async done => {
+            randomUser = {
+                firstName: faker.name.firstName(),
+                lastName: faker.name.lastName(),
+                email: faker.internet.email(),
+                password: faker.internet.password()
+            };
+            registerResponse = await request(app).post('/api/users/register')
+                .type('form')
+                .send({...randomUser})
+                .set('Accept', 'application/json');
+            done()
+        });
+
+        test('user should be able to login', async () => {
+            const {email, password} = randomUser;
+            const response = await request(app).post('/api/users/login')
+                .type('form')
+                .send({email, password})
+                .set('Accept', 'application/json');
+            expect(response.body.msg).toContain('logged in');
+            expect(response.body.token).toContain('Bearer');
+        });
+
+        test('user should not be able to update data when he have not confirmed email', async () => {
```

Comment:
`user should not be able to update data when he have not confirmed email`

this test failed

Reply:
@damian0o separating routes might be a overkill i think. Because now we just did 
`module.exports = app` from app.js and then consume it by our server.js and tests.
I'v we want to separate routes from app we have to export app once more without attached routes but only witch attached basic things like 
```
app.use(express.json());
app.use(express.urlencoded({ extended: false }));
```
And then add all routes in server.js version and attach only needed route in test version
I dont see a clean way to do this. 
Maybe we can create create a new file.js where we declare our testing app once more with only attached needed middlewares and then use it only for testing and use old app.js only with server.js but I don't think that some more declared routes make that huge effort that we need to do things this way.